ADApp/ADSrc/NDArray.h

```diff
@@ -124,6 +126,26 @@ class epicsShareClass NDArray {
     NDAttributeList *pAttributeList;  /**< Linked list of attributes */
 };
 
+// This class defines the object that is contained in the std::multilist for sorting NDArrays in the freeList_.
+// It defines the < operator to use the NDArray::dataSize field as the sort key
+
+// We would like to hide this class definition in NDArrayPool.cpp and just forward reference it here.
+// That works on Visual Studio, and on gcc if instantiating plugins as heap variables with "new", but fails on gcc
+// if instantiating plugins as automatic variables.
+//class sortedListElement;
+
+class freeListElement {
+    public:
+        freeListElement(NDArray *pArray, size_t dataSize) {
+          pArray_ = pArray;
+          dataSize_ = dataSize;}
+        friend bool operator<(const freeListElement& lhs, const freeListElement& rhs) {
+            return (lhs.dataSize_ < rhs.dataSize_);
+        }
+        NDArray *pArray_;
+        size_t dataSize_;
+};
```

Comment:
Discussed assignment operator and copy constructor but not necessary as the private members include a simple pointer and standard size_t type; default compiler generated methods will work.